packages/web3-eth-abi/src/api/functions_api.ts

```diff
@@ -0,0 +1,41 @@
+import { AbiError } from 'web3-common';
+import { sha3Raw } from 'web3-utils';
+import { isAbiFunctionFragment, jsonInterfaceMethodToString } from '../utils';
+import { JsonAbiFunctionFragment } from '../types';
+import { encodeParameters } from './parameters_api';
+
+/**
+ * Encodes the function name to its ABI representation, which are the first 4 bytes of the sha3 of the function name including  types.
+ */
+export const encodeFunctionSignature = (functionName: string | JsonAbiFunctionFragment): string => {
+	if (typeof functionName !== 'string' && !isAbiFunctionFragment(functionName)) {
+		throw new Error('Invalid parameter value in encodeFunctionSignature');
+	}
+
+	let name: string;
+
+	if (typeof functionName === 'function' || (typeof functionName === 'object' && functionName)) {
```

Comment:
using advantage of short circut &&

```suggestion
	if ( functionName && ( typeof functionName === 'function' || typeof functionName === 'object')) {
```